pydis_site/apps/staff/viewsets/logs.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+from django.core.handlers.wsgi import WSGIRequest
+from django.http import HttpResponse
+from django.shortcuts import get_object_or_404, render
+from django.views import View
+
+from pydis_site.apps.api.models.bot.message_deletion_context import MessageDeletionContext
+
+log = logging.getLogger(__name__)
+
+
+class LogView(View):
+    """The default view for the Deleted Messages logs."""
+
+    template_name = "staff/logs.html"
+
+    def get(self, request: WSGIRequest, pk: int) -> HttpResponse:
+        """Get method that answers a request with an html response by rendering a template."""
+        message_context = get_object_or_404(MessageDeletionContext, pk=pk)
+
+        actor = message_context.actor
+        creation = message_context.creation
+        messages = message_context.deletedmessage_set.all()
+
+        template_fields = {
+            'actor': actor,
+            'actor_colour': message_context.actor.top_role.colour,
+            'creation': creation,
+            'messages': messages
+        }
+
+        return render(request, self.template_name, template_fields)
```

Comment:
This should use a generic view, specifically `DetailView`. You can add extra template context if needed as shown here: https://docs.djangoproject.com/en/2.2/topics/class-based-views/generic-display/#adding-extra-context , although from what I can tell you don#t need that here in the first place (just the context will be enough)

Reply:
Thanks, I'll change this.

One of the reasons I created custom template fields was to avoid the deep nesting the messages had, but since I only refer to it once in the template in the `for`-loop, it's probably better not to do that and just go with the default context. That makes maintaining this a lot easier and reduces the view to just a couple of lines.